conformance/packages/dns-test/src/tshark.rs

```diff
@@ -20,23 +21,24 @@ impl Container {
     pub fn eavesdrop(&self) -> Result<Tshark> {
         let id = ID.fetch_add(1, atomic::Ordering::Relaxed);
         let pidfile = pid_file(id);
-        let capture_file = capture_file(id);
 
-        // `docker exec $child` merges the child's stderr and stdout streams and pipes them into
-        // its stdout. as we cannot tell stdout (JSON) from stderr (log message) from the host side,
-        // we'll redirect the JSON output to a file inside the container and read the log messages
-        // from the host side
-        // --log-level info --log-domain main
         let tshark = format!(
             "echo $$ > {pidfile}
-exec tshark --log-level debug --log-domain main,capture -l -i eth0 -T json -O dns -f 'udp port {UDP_PORT}' > {capture_file}"
+exec tshark --log-level debug --log-domain main,capture -l -i eth0 -T json -O dns -f 'udp port {UDP_PORT}'"
         );
         let mut child = self.spawn(&["sh", "-c", &tshark])?;
 
-        let stdout = child.stdout()?;
-        let mut stdout = BufReader::new(stdout).lines();
+        let stderr = child.stderr()?;
+        let mut stderr = BufReader::new(stderr).lines();
 
-        for res in stdout.by_ref() {
+        let mut stdout = child.stdout()?;
+        let stdout_handle = thread::spawn(move || -> CoreResult<String, io::Error> {
```

Comment:
at a glance I don't see why a new thread is needed if we are not going to try to concurrently read from both stdout and stderr from the same thread. the `terminate` method reads from stdout only *after* it's done with stderr.

Reply:
I tried moving reading stdout to the same thread, after reading stderr, and this caused a deadlock in `tshark::tests::resolver`. This is because the stdout pipe buffer may fill up, which blocks the child process, and prevents it from writing the packet-related messages to stderr that we are waiting for. (I'm not sure if this is blocking the `docker exec` process or the `tshark` process itself, but the effect is the same either way) Reading stdout into an unlimited buffer on another thread avoids this blocking, and lets us keep reading stderr messages in each `wait_for_capture()` or `terminate()` call.

Reply:
I see. thanks for the explanation